Makefile

```diff
@@ -14,10 +14,25 @@ CMAKE_EXTRA_FLAGS ?=
 # CMAKE_INSTALL_PREFIX
 #   - May be passed directly or as part of CMAKE_EXTRA_FLAGS.
 #   - `checkprefix` target checks that it matches the CMake-cached value. #9615
+ifeq (,$(CMAKE_EXTRA_FLAGS))
+CMAKE_INSTALL_PREFIX ?=
+else
 CMAKE_INSTALL_PREFIX ?= $(shell echo $(CMAKE_EXTRA_FLAGS) | 2>/dev/null \
     grep -o 'CMAKE_INSTALL_PREFIX=[^ ]\+' | cut -d '=' -f2)
+endif
```

Comment:
why is this extra stuff needed? line 20 should already set `CMAKE_INSTALL_PREFIX` as needed.

Reply:
Which extra stuff?
This avoids the `$(shell)` for when no extra flags are used.  A small optimization, but just more simple when not needing the complexity (I've noticed all this initially with `make -n`).

Reply:
How is it simpler? It is more code. Not worth the noise.

Reply:
It is less processing in general (no shell), and more silent with `make -n`.
If could also be surrounded by just `ifneq (,$(CMAKE_EXTRA_FLAGS))`, but I've thought this would indicate better that it is meant to come from outside.

Reply:
Changed it.

Reply:
There is no need to optimize  for `make -n` at the cost of extra code. The shell command performance cost is negligible. Code readability dominates the priorities here, the Makefile is already too big.